File: ClinicSystem/src/com/example/clinicsystem/models/classes/M_Employee.java
package com.example.clinicsystem.models.classes;

import com.example.clinicsystem.helpers.DatabaseConnection;
import com.example.clinicsystem.models.enums.Gender;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class M_Employee extends M_Person {
    private int employeeID;
    private String email;

    public M_Employee(String firstName, String middleName, String lastName, String phoneNumber, String birthDate, M_Address address, int nationalID, Gender gender, int employeeID, String email) {
        super(firstName, middleName, lastName, phoneNumber, birthDate, address, nationalID, gender);
        this.employeeID = employeeID;
        this.email = email;
    }

    public M_Employee() {
        super();
    }

    public List<M_Employee> getAllEmployees(String condition) {
        List<M_Employee> employees = new ArrayList<>();
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();
        ResultSet resultSet = databaseConnection.select("employee", condition);

        if(resultSet != null) {
            try {
                while (resultSet.next()) {
                    int id = resultSet.getInt("Employee_ID");
                    String email = resultSet.getString("Employee_Email");

                    List<M_Person> people = getAllPersons(" WHERE Person_ID = " + resultSet.getInt("Person_ID"));

                    M_Employee employee = new M_Employee(people.get(0).getFirstName(), people.get(0).getMiddleName(),
                            people.get(0).getLastName(), people.get(0).getPhoneNumber(), people.get(0).getBirthDate(),
                            people.get(0).getAddress(), people.get(0).getNationalID(), people.get(0).getGender(),
                            id, email);

                    employees.add(employee);
                }
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

        return employees;
    }

    public void addEmployee(int person) {
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();

        databaseConnection.insertInto("employee","(Employee_ID, Person_ID, Employee_Email)" ,"('" + this.employeeID + "', '"
                + person + "', '" + this.email + "')");
    }

    public int getEmployeeID() {
        return employeeID;
    }

    public void setEmployeeID(int employeeID) {
        this.employeeID = employeeID;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: clinic/M_Employee.php
<?php
include_once "M_Person.php";

class M_Employee extends M_Person {
	private $employeeID;
    private $email;
    private $firstName, $middleName, $lastName, $phoneNumber, $birthDate,
		$address, $nationalID, $gender;
    public function __construct($firstName, $middleName, $lastName, $phoneNumber, $birthDate,
		$address, $nationalID, $gender, $employeeID, $email) {
        $this->firstName =   $firstName;
		$this->middleName =  $middleName;
		$this->lastName =    $lastName;
		$this->phoneNumber = $phoneNumber;
		$this->birthDate =   $birthDate;
		$this->address =     $address;
		$this->nationalID =  $nationalID;
		$this->gender = 	 $gender;
        $this->employeeID =  $employeeID;
        $this->email      =  $email;
    }

    public function getAllEmployees($condition) {
        $conn = DatabaseConnection::getINSTANCE();
		
		$result = $conn->select("employee", $condition);
		$employees = array();
		
		while ($row = $result->fetch_assoc()) {
			$people = $this->getAllPersons(" WHERE Person_ID = " . 
				$row["Person_ID"]);
			$addresses = $this->getAllAddresses(" WHERE Address_ID = " . 
				$people[0]->getAddress()->getID());
			
			$employee = new M_Employee($people[0]->getFirstName(), $people[0]->getMiddleName()
				, $people[0]->getLastName(),$people[0]->getPhoneNumber(), $people[0]->getBirthDate()
				, $addresses[0], $people[0]->getNationalID(), $people[0]->getGender()
				, $row["Employee_ID"], $row["Employee_Email"]);
			array_push($employees, $employee);
		}
		
		return $employees;
    }

    public function addEmployee($person) {
        $conn = DatabaseConnection::getINSTANCE();
		
		return $conn->insertInto("employee", "(Employee_ID, Person_ID, Employee_Email)", "('". 
		$this->employeeID . "', '" . $person . "', '" . $this->email . "')");
    }

    public function getEmployeeID() {
        return $this->employeeID;
    }

    public function setEmployeeID($employeeID) {
        $this->employeeID = $employeeID;
    }

    public function getEmail() {
        return $this->email;
    }

    public function setEmail($email) {
        $this->email = $email;
    }

    public function getFirstName() {
        return $this->firstName;
    }

    public function setFirstName($firstName) {
        $this->firstName = $firstName;
    }

    public function getMiddleName() {
        return parent::getMiddleName();
    }

    public function setMiddleName($middleName) {
        parent::setMiddleName($middleName);
    }

    public function getLastName() {
        return parent::getLastName();
    }

    public function setLastName($lastName) {
        parent::setLastName($lastName);
    }

    public function getPhoneNumber() {
        return parent::getPhoneNumber();
    }

    public function setPhoneNumber($phoneNumber) {
        parent::setPhoneNumber($phoneNumber);
    }

    public function getBirthDate() {
        return parent::getBirthDate();
    }

    public function setBirthDate($birthDate) {
        parent::setBirthDate($birthDate);
    }

    public function getAddress() {
        return parent::getAddress();
    }

    public function setAddress($address) {
        parent::setAddress($address);
    }

    public function getNationalID() {
        return parent::getNationalID();
    }

    public function setNationalID($nationalID) {
        parent::setNationalID($nationalID);
    }

    public function getGender() {
        return parent::getGender();
    }

    public function setGender($gender) {
        $this->gender = $gender;
    }
}

?>